packages/arb-validator/structures/node.go

```
@@ -134,8 +130,8 @@ func NewRandomInvalidNodeFromValidPrev(
 		common.NewTimeBlocks(common.RandBigInt()),
 	)
 
-	_ = nextNode.UpdateInvalidOpinion()
-	return nextNode
+	err := nextNode.UpdateInvalidOpinion()
```

I think we should be able to safely ignore this error. Instead it'd be good to add the `kind != valprotocol.ValidChildType` check into this function and panic if it's violated. This is only used in tests so it's nice to keep the API simple

---

done

---

added appropriate panic